tests/Doctrine/Tests/DBAL/Functional/Driver/SQLAnywhere/StatementTest.php

```diff
@@ -50,4 +51,54 @@ public function testPersistentStatement()
         self::assertTrue($prepStmt->execute(),' Statement persistent failed');
     }
 
+    /**
+     * Check Issue #2991
+     * @throws \Doctrine\DBAL\ConnectionException
+     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
+     */
+    public function testBindParams()
+    {
+        $params = $this->_conn->getParams();
+
+        $conn = DriverManager::getConnection($params);
+
+        $conn->connect();
+        $conn->beginTransaction();
+        try {
+            self::assertTrue($conn->isConnected(),'No SQLAnywhere-Connection established');
+
+            $sql = 'DROP TABLE IF EXISTS testTable2991';
+            $prepStmt = $conn->prepare($sql);
+            $prepStmt->execute();
+
+            $sql = 'CREATE TABLE IF NOT EXISTS testTable2991(
+                      col1 VARCHAR(80 CHAR) NOT NULL,
+                      col2 VARCHAR(20) NOT NULL,
+                      col3 INTEGER NOT NULL
```

Comment:
Are three columns required to reproduce the case or fewer could be used?

Reply:
I reduced it to 2 column, a varchar and an int. Only with this preparation the error
```
Caused by
Doctrine\DBAL\Driver\SQLAnywhere\SQLAnywhereException: SQLSTATE [00000] [-158] Value 139645189756408 out of range for destination
```
can be reproduced.
The driver access memory, which is freed